lib/storage/tree.rb

```diff
@@ -0,0 +1,54 @@
+require "tty-tree"
+
+module Storage
+  class Tree
+    attr_reader :name, :children
+  
+    def initialize(name, children)

```

Comment:
What kind of format is `initialize` expecting `children` to be?

Reply:
Currently the format of `children` is literally just an array of strings and tree objects, you're essentially just building the tree yourself. A function to allow a user to initialise a tree with (e.g.) a hash is probably a nice feature to add in future

Reply:
I think including the functionality to construct a `Tree` and any nested `Trees` based off of a hash input is best included in this body of work; that's how we'll be constructing `Trees` from the output from cloud providers. You should assume here that the input for `children` will always be of the form:
```ruby
[
  "file1",
  "file2",
  {
    "dir1" => [
      "file3"
    ]
  }
]
```

You'll note that each sub-hash in the above hash matches the format of `label => children`. This form lends itself to recursive `Tree` construction fairly simply. I posted a quick attempt at this in the Slack channel, if you'd like a reference.